src/classes/RD2_ApiService.cls

```diff
@@ -0,0 +1,181 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description All services specific to Enhanced Recurring Donation Api's
+*
+*/
+public inherited sharing class RD2_ApiService {
+
+    /**
+     * @description Retrieve a map of RecurringDonationSchedule__c records by Rd.id using an inner wrapper
+     * class representation of the RdSchedule object because it's protected and customers will not be able to
+     * deserialize it.
+     * @param rdIds Recurring Donation Id's to retrieve schedules for
+     * @return JSON Serialized version of Map<Id, List<RD2_ApiService.Schedule>> schedulesByRdId
+     */
+    public String querySchedules(Set<Id> rdIds) {
+        Map<Id, List<RD2_ApiService.Schedule>> schedulesByRdId = new Map<Id, List<RD2_ApiService.Schedule>>();
+
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
+            return JSON.serialize(schedulesByRdId);
+        }
+
+        RD2_ScheduleService.PauseScheduleHandler pauseHandlerService = new RD2_ScheduleService.PauseScheduleHandler();
+
+        String soql = getRecurringDonationQuery();
+        for (npe03__Recurring_Donation__c rd : Database.query(soql)) {
+
+            Boolean isRdCurrentlyPaused = pauseHandlerService.isCurrentlyPaused(
+                rd.RecurringDonationSchedules__r, RD2_ScheduleService.currentDate
+            );
+
+            List<RD2_ApiService.Schedule> schedules = new List<RD2_ApiService.Schedule>();
+            for (RecurringDonationSchedule__c s : rd.RecurringDonationSchedules__r) {
+                RD2_ApiService.Schedule schedule = new RD2_ApiService.Schedule(s, rd);
+                if (isRdCurrentlyPaused && schedule.isActiveAndNotPaused()) {
```

Comment:
I think we just need to check `if (isRdCurrentlyPaused  && schedule.isCurrent && !schedule.isPause) {` in order to set the `isCurrent` to false. We might not need a new method for this one line.